### PR TITLE
Add promises add fetch for isolated instance

### PR DESCRIPTION
**Pull request by Nathan219**
This adds the ability for isolated containers to default (during dns) to their isolated brethren instead of just master

Add promises
add fetch for isolated instance
#### Dependencies
- [x] CodeNow/api#1511
#### Reviewers
- [x] @anandkumarpatel

### CODE REVIEW

**Comment by bkendall** [comment]
I don't see anything wrong with this... lgtm

File: lib/models/user.js [hunk 1]
```diff
@@ -18,6 +19,8 @@ var find = require('101/find');
 var hasProps = require('101/has-properties');
 var exists = require('101/exists');
 var qs = require('querystring');
+var Promise = require('bluebird');
+var createCount = require('callback-count')
```
Comment (bkendall) [inline lib/models/user.js]: `callback-count` isn't used?

Comment (bkendall) [inline lib/models/user.js]: (remove it from `package.json` as well)

**Comment by bkendall** [comment]
Think you have some tests to look at :)

File: package.json [hunk 2]
```diff
@@ -35,7 +35,9 @@
   "dependencies": {
     "101": "^1.1.0",
     "@runnable/hostname": "^3.0.0",
+    "bluebird": "^3.3.5",
     "boom": "~2.4.1",
+    "callback-count": "^0.2.0",
```
Comment (bkendall) [inline package.json]: I don't see this being used

**Comment by bkendall** [comment]
LGTM

**Comment by bkendall** [comment]
One question in there about an extra require. Also, does this end up changing current behavior? Looks to me like it does, but want to be sure

**Comment by Nathan219** [comment]
This keeps current behavior; I just made fetchBackendUrl match similarly to fetchInternalIpForHostname, though I don't think anything uses that function at all (after doing a search on github, I don't see any)

File: lib/models/user.js [hunk 3]
```diff
@@ -186,40 +188,40 @@ User.prototype.oauthId = function () {
  */
 User.prototype.fetchInternalIpForHostname = function (hostname, containerLocalIp, localDockHost, cb) {
   var self = this;
-    // masterInstance exists
-    var refererQuery = {
-      'container.inspect.NetworkSettings.IPAddress': containerLocalIp,
-      'container.dockerHost': localDockHost
-    };
-    self._fetchInstanceAndDepWithHostname(refererQuery, hostname, function (err, dep) {
+  // masterInstance exists
```
Comment (Nathan219) [inline lib/models/user.js:191]: fixed the layout here

**Comment by anandkumarpatel** [comment]
- [x] @anandkumarpatel

**Pull request merged**